scipy/conftest.py

```diff
@@ -171,3 +171,15 @@ def wrapped(*args, xp, **kwargs):  # type: ignore[misc]
                 return func(*args, xp, **kwargs)
         return wrapped
     return wrapper
+
+
+def set_assert_allclose(xp=None):
+    if xp is None:
+        return npt.assert_allclose
+    if 'cupy' in xp.__name__:
+        import cupy as cp
+        return cp.testing.assert_allclose
+    elif 'torch' in xp.__name__:
+        import torch
+        return torch.testing.assert_close
```

Comment:
If I'm not mistaken, we might be able to avoid the extra imports and do `return xp.testing...`.

Reply:
Yep, just tested and that works fine thanks, will push a change.